bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/type/AbstractChannelTypeBuilder.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.core.thing.type;
+
+import java.net.URI;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * Abstract base class with common methods for {@link ChannelTypeBuilder}
+ *
+ * @author Stefan Triller - Initial contribution
+ *
+ */
+@NonNullByDefault
+abstract class AbstractChannelTypeBuilder<T extends IChannelTypeBuilder<T>> implements IChannelTypeBuilder<T> {
+
+    protected final ChannelTypeUID channelTypeUID;
+    protected final String label;
+    protected boolean advanced;
+    protected @Nullable String description;
+    protected @Nullable String category;
+    protected final Set<String> tags = new HashSet<>();
+    protected @Nullable URI configDescriptionURI;
+
+    public AbstractChannelTypeBuilder(ChannelTypeUID channelTypeUID, String label) {
+        this.channelTypeUID = channelTypeUID;
+        this.label = label;
```

Comment:
in a previous version both parameters where checked for empty-ness, would be nice to see it again.